crates/ink/codegen/src/generator/trait_def/call_builder.rs

```diff
@@ -195,6 +195,20 @@ impl CallBuilder<'_> {
                         .finish()
                 }
             }
+
+            #[cfg(feature = "std")]
+            /// We require this manual implementation since the derive produces incorrect trait bounds.
+            impl<E> ::scale_info::TypeInfo for #call_builder_ident<E>
+            where
+                E: ::ink::env::Environment,
+                <E as ::ink::env::Environment>::AccountId: ::scale_info::TypeInfo + 'static,
+            {
+                type Identity = <E as ::ink::env::Environment>::AccountId;
```

Comment:
Hm I guess we can't do `type Identity = #call_builder_ident<E>` here because it will complain about `'static` lifetime on `E`?

Downside is the moment there is no way to tell the difference between a plain `AccountId` and this in the contract metadata. Not the end of the world though, as it is still an accurate representation of the actual encoded data.

Reply:
I did it because I didn't want the appearance of the `__ink_CallBuilderFlipper` in the metadata=D After all, it would look weird to the user. Plus, it is possible to have several different `__ink_CallBuilderFlipper`(when the user defines traits with namespaces or in the different modules), and it also may confuse.

Reply:
Yes makes sense.